### PR TITLE
[WIP] cli dev server output

### PR DESCRIPTION
**Pull request by danielterwiel**

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -174,13 +185,11 @@ fn start_server(
...
 ) -> (
     LSPClient,
-    UnboundedSender<Vec<u8>>,
-    UnboundedSender<Vec<u8>>,
     impl Future<Output = ()>,
 ) {
     let cargo = env::var("CARGO").unwrap_or_else(|_| "cargo".to_string());
     let mut process = Command::new(cargo)
-        .env("RUST_LOG", "trace")
+        .env("RUST_LOG", "info")
```
Comment (SuddenlyHazel) [inline cli/src/main.rs:190]: changed this back to info so we don't spam the pipes

File: cli/src/main.rs [hunk 2]
```diff
@@ -199,78 +207,55 @@ fn start_server(
...
-                    info!("rx_recv got: {:?}", String::from_utf8(data));
-
-                    let mut buf = String::new();
-                    let _ = reader.read_line(&mut buf);
-                }
-                None => {
-                    debug!("rx_recv: quitting loop");
-                    break;
-                }
+            match reader.next_line().await {
```
Comment (SuddenlyHazel) [inline cli/src/main.rs:232]: Directly reading out the lines from the stream here.

File: xtask/src/main.rs [hunk 3]
```diff
@@ -188,7 +188,7 @@ fn build_component(guest: &Guest) -> Result<(), DynError> {
         &dir_components().display(),
         guest.name
     );
-
+    print!("{} {} {}", cmd_wasm_guest, cmd_wasi_snapshot, cmd_component_output);
```
Comment (SuddenlyHazel) [inline xtask/src/main.rs]: Will rm this

File: cli/src/main.rs [hunk 4]
```diff
@@ -279,6 +264,8 @@ async fn start(
     rpc_port: &u16,
     open_browser: &bool,
 ) -> Result<(), DynError> {
+    info!("Attempting to start servers");
```
Comment (SuddenlyHazel) [inline cli/src/main.rs:265]: will rm this

File: cli/src/main.rs [hunk 5]
```diff
@@ -144,6 +157,105 @@ async fn poll_server_listening() -> Result<(), DynError> {
...
+async fn trigger(domain: &String, http_port: &u16, rpc_port: &u16, open_browser: &bool) {
+    let (client, wait) = start_server(domain, http_port, rpc_port, open_browser);
+
+    // Spin off child process to make sure it can make process on its own
+    // while we read its output
+
+    let mut process = client.process;
+
+    // todo tokio select on this
+    let task_handle = tokio::spawn(async move {
```
Comment (SuddenlyHazel) [inline cli/src/main.rs:169]: If this `task_handle` returns we need to abort. it's indicative the task shut down.

Comment (danielterwiel) [reply cli/src/main.rs:169]: I suspect this solves another possible scenario I still needed to test:

running `cargo run stop` while the development_server is running should terminate the development_server process

🙌

**Pull request merged**